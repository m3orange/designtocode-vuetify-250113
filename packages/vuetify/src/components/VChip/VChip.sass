// Imports
@import '../../styles/styles.sass'
@import './_mixins.sass'
@import './_variables.scss'

// Theme
.v-chip:not(.v-chip--outlined)
  &.primary,
  &.secondary,
  &.accent,
  &.success,
  &.error,
  &.warning,
  &.info
    color: map-deep-get($material-dark, 'text', 'primary')

.v-chip.theme--light
  +chip-theme($material-light)
.v-chip.theme--dark
  +chip-theme($material-dark)

// Block
.v-chip
  align-items: center
  cursor: pointer
  display: inline-flex
  line-height: 20px
  outline: none
  padding: 0 12px
  position: relative
  transition-duration: 0.28s
  transition-property: box-shadow, opacity
  transition-timing-function: map-get($transition, 'fast-out-slow-in')
  vertical-align: middle

  &:after
    background-color: currentColor
    bottom: 0
    border-radius: inherit
    content: ''
    left: 0
    opacity: 0
    position: absolute
    pointer-events: none
    right: 0
    top: 0
    transition: opacity 0.2s map-get($transition, 'ease-in-out')

  &:active
    +elevation(2)

  &:focus:after
    opacity: .22

  .v-avatar
    height: 24px !important
    width: 24px !important

  .v-icon
    font-size: 24px

  .v-avatar,
  .v-icon
    &--left
      margin-left: -6px
      margin-right: 8px

    &--right
      margin-left: 8px
      margin-right: -6px

// Elements
.v-chip__close.v-icon
  font-size: 18px
  max-height: 18px
  max-width: 18px
  user-select: none

  &.v-icon--right
    margin-right: -4px

  &:hover,
  &:focus,
  &:active
    opacity: .72

.v-chip__content
  align-items: center
  display: inline-flex
  height: 100%

// Modifiers
.v-chip--disabled
  opacity: .4
  pointer-events: none

.v-chip--draggable
  &:active
    +elevation(12)

.v-chip--clickable
  cursor: pointer

.v-chip--label
  border-radius: 4px

.v-chip.v-chip--outlined
  background-color: transparent !important
  border: $icon-outlined-border-width solid currentColor

  .v-icon
    color: inherit

.v-chip
  @each $name, $size in $icon-sizes
    &.v-size--#{$name}
      border-radius: #{map-get($size, 'height') / 2}px
      font-size: #{map-get($size, 'font-size')}px
      height: #{map-get($size, 'height')}px

// v-chip($material)
//   background: $material.chips.background
//   color: $material.chips.color

//   &--disabled
//     color: $material.text.disabled

// v-chip-rtl()
//   &__close
//     margin: $chip-close-margin-rtl

//   &--removable
//     .v-chip__content
//       padding: $chip-removable-padding-rtl

//   &--select-multi
//     margin: $chip-margin 0 $chip-margin $chip-margin

//   .v-avatar
//     margin-right: -12px
//     margin-left: 8px

//   .v-icon
//     &--right
//       margin-right: $chip-icon-offset
//       margin-left: $chip-icon-negative-offset

//     &--left
//       margin-right: $chip-icon-negative-offset
//       margin-left: $chip-icon-offset

// theme(v-chip, "v-chip")
// rtl(v-chip-rtl, "v-chip")

// .v-chip
//   align-items: center
//   border-radius: $chip-border-radius
//   display: inline-flex
//   font-size: $chip-font-size
//   margin: $chip-margin
//   outline: none
//   position: relative
//   transition: $primary-transition
//   vertical-align: middle

//   .v-chip__content
//     align-items: center
//     border-radius: $chip-border-radius
//     cursor: default
//     display: inline-flex
//     height: $chip-height
//     justify-content: space-between
//     padding: $chip-padding
//     vertical-align: middle
//     white-space: nowrap
//     z-index: 1

//   &--removable
//     .v-chip__content
//       padding: $chip-removable-padding

//   .v-avatar
//     // Important is needed to account for new avatar structure
//     height: $chip-height !important
//     margin-left: -12px
//     margin-right: 8px
//     min-width: $chip-height
//     width: $chip-height !important

//     img
//       height: 100%
//       width: 100%

//   &:focus:not(.v-chip--disabled),
//   &--active,
//   &--selected
//     border-color: rgba(#000, .13)
//     // overflow: hidden TEMP
//     elevation(2)

//     &:after
//       background: currentColor
//       border-radius: inherit
//       content: ''
//       height: 100%
//       position: absolute
//       top: 0
//       left: 0
//       transition: inherit
//       width: 100%
//       pointer-events: none
//       opacity: .13

//   &--label
//     border-radius: $chip-label-border-radius

//     .v-chip__content
//       border-radius: $chip-label-border-radius

//   // Increase specificity to override theme background
//   &\&.v-chip--outline
//     background: $chip-outline-background
//     border: 1px solid currentColor
//     color: $chip-outline-color
//     height: $chip-height

//     .v-avatar
//       margin-left: -13px

//   &--small
//     height: $chip-small-height !important

//     .v-avatar
//       height: $chip-small-height !important
//       min-width: $chip-small-height
//       width: $chip-small-height !important

//     .v-icon
//       font-size: $chip-small-font-size

//   &__close
//     align-items: center
//     color: inherit
//     display: flex
//     font-size: $chip-icon-font-size
//     margin: $chip-close-margin
//     text-decoration: none
//     user-select: none

//     > .v-icon
//       color: inherit !important
//       font-size: 20px
//       cursor: pointer
//       opacity: .5

//       &:hover
//         opacity: 1

//   &--disabled
//     .v-chip__close
//       pointer-events: none

//   &--select-multi
//     margin: $chip-margin $chip-margin $chip-margin 0

//   .v-icon
//     color: inherit

//     &--right
//       margin-left: $chip-icon-offset
//       margin-right: $chip-icon-negative-offset

//     &--left
//       margin-left: $chip-icon-negative-offset
//       margin-right: $chip-icon-offset
