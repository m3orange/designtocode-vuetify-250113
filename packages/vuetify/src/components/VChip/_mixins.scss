@mixin chip-sizes {
  @each $sizeName, $multiplier in $size-scales {
    $size: map-get($chip-sizes, "font-size") + (2 * $multiplier) / 16;
    $height: map-get($chip-sizes, "height") + (8 * $multiplier);

    .v-chip--size-#{$sizeName} {
      --v-chip-size: #{$size};
      --v-chip-height: #{$height};
      font-size: $size;

      .v-avatar {
        --v-avatar-height: #{$height - $multiplier}
      }
    }
  }
}

@mixin chip-density($properties, $densities) {
  @each $density, $multiplier in $densities {
    $value: calc(var(--v-chip-height) + #{$multiplier * $spacer});

    &.v-chip--density-#{$density} {
      @if type-of($properties) == "list" {
        @each $property in $properties {
          #{$property}: $value;
        }
      } @else {
        #{$properties}: $value;
      }
    }
  }
}
