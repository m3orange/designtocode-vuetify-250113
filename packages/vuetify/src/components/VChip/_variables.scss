// Defaults
$chip-background-color: rgba(
  var(--v-border-color),
  var(--v-border-opacity)
) !default;
$chip-border: (
  $chip-border-color,
  $chip-border-style,
  $chip-border-width
) !default;
$chip-border-color: $border-color-root !default;
$chip-border-radius: map-get($rounded, "pill") !default;
$chip-border-style: $border-style-root !default;
$chip-border-thin-width: thin !default;
$chip-border-width: 0 !default;
$chip-close-size: 18px !default;
$chip-density: (
  "default": 0,
  "comfortable": -2,
  "compact": -3,
) !default;
$chip-disabled-opacity: 0.3 !default;
$chip-elevation: 0 !default;
$chip-filter-max-width: 72px !default;
$chip-font-size: map-deep-get($typography, "button", "size") !default;
$chip-height: 36px !default;
$chip-icon-density: (
  "default": 3,
  "comfortable": 0,
  "compact": -2,
) !default;
$chip-icon-font-size: 24px !default;
$chip-icon-margin-before: 4px !default;
$chip-icon-margin-after: 4px !default;
$chip-icon-right-margin-after: 4px !default;
$chip-icon-size: 48px !default;
$chip-icon-sizes: () !default;
$chip-icon-sizes: map-merge(
  (
    "font-size": $chip-icon-font-size,
    "size": $chip-icon-size,
  ),
  $chip-icon-sizes
);
$chip-label-border-radius: 47% !default;
$chip-link-focus-opacity: 0.3 !default;
$icon-outlined-border-width: 2px !default;
$icon-outlined-active-opacity: 0.6 !default;
$chip-padding-ratio: 2.25 !default;
$chip-pill-filter-margin: 4px !default;
$chip-pill-avatar-size: 18px !default;
$chip-pill-avatar-margin-before: 4px !default;
$chip-rounded-border-radius: $border-radius-root !default;
$chip-selected-opacity: 0.8 !default;
$chip-sizes: () !default;
$chip-sizes: map-merge(
  (
    "height": $chip-height,
    "font-size": $chip-font-size,
    "width-ratio": $chip-width-ratio,
    "padding-ratio": $chip-padding-ratio,
  ),
  $chip-sizes
);
$chip-stacked-density: (
  "default": 0,
  "comfortable": -4,
  "compact": -6,
) !default;
$chip-width-ratio: 16/9 !default;
$primary-transition: 1s !default;
