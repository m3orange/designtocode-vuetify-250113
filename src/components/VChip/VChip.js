require('../../stylus/components/_chips.styl')

import VIcon from '../VIcon'
import Colorable from '../../mixins/colorable'
import Themeable from '../../mixins/themeable'
import Toggleable from '../../mixins/toggleable'

export default {
  name: 'v-chip',

  components: {
    VIcon
  },

  mixins: [Colorable, Themeable, Toggleable],

  props: {
    close: Boolean,
    disabled: Boolean,
    label: Boolean,
    outline: Boolean,
    // Used for selects/tagging
    selected: Boolean,
    selectedClass: {
      type: String,
      default: 'accent'
    },
    small: Boolean,
    value: {
      type: Boolean,
      default: true
    }
  },

  computed: {
    classes () {
      return {
        'chip': true,
        'chip--disabled': this.disabled,
        [`chip--selected ${this.selectedClass}`]: this.selected,
        'chip--label': this.label,
        'chip--outline': this.outline,
        'chip--small': this.small,
        'chip--removable': this.close
      }
    }
  },

  render (h) {
    const children = [this.$slots.default]
    const data = {
      'class': this._computedClasses,
      attrs: { tabindex: this.disabled ? -1 : 0 },
      directives: [{
        name: 'show',
        value: this.isActive
      }],
      on: this.$listeners
    }

    if (this.close) {
      const data = {
        staticClass: 'chip__close',
        on: {
          click: e => {
            e.stopPropagation()

            this.$emit('input', false)
          }
        }
      }

      children.push(h('div', data, [
        h(VIcon, { props: { right: true } }, 'cancel')
      ]))
    }

    return h('span', data, children)
  }
}
