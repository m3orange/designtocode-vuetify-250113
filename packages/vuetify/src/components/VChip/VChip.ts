import './VChip.sass'

// Types
import { VNode } from 'vue'
import mixins from '../../util/mixins'

// Components
import VIcon from '../VIcon'

// Mixins
import Colorable from '../../mixins/colorable'
import Themeable from '../../mixins/themeable'
import Toggleable from '../../mixins/toggleable'
import Sizeable from '../../mixins/sizeable'

// Directives
import Ripple from '../../directives/ripple'

// Utiles
import { deprecate } from '../../util/console'

/* @vue/component */
export default mixins(
  Colorable,
  Sizeable,
  Themeable,
  Toggleable
).extend({
  name: 'v-chip',

  directives: { Ripple },

  props: {
    close: Boolean,
    disabled: Boolean,
    draggable: Boolean,
    label: Boolean,
    outline: Boolean,
    outlined: Boolean,
    ripple: {
      type: [Boolean, Object],
      default: null
    },
    // Used for selects/tagging
    selected: Boolean,
    textColor: String,
    value: {
      type: Boolean,
      default: true
    }
  },

  computed: {
    classes (): object {
      return {
        'v-chip--disabled': this.disabled,
        'v-chip--draggable': this.draggable,
        'v-chip--selected': this.selected && !this.disabled,
        'v-chip--label': this.label,
        'v-chip--outlined': this.hasOutline,
        'v-chip--removable': this.close,
        'v-chip--clickable': this.$listeners.click || this.$listeners['!click'],
        ...this.themeClasses,
        ...this.sizeableClasses
      }
    },
    hasOutline (): boolean {
      return this.outline || this.outlined
    }
  },

  methods: {
    genClose (): VNode {
      return this.$createElement(VIcon, {
        staticClass: 'v-chip__close',
        props: {
          right: true
        },
        on: {
          click: (e: Event) => {
            e.stopPropagation()

            this.isActive = false
          }
        }
      }, '$vuetify.icons.delete')
    },
    genContent (): VNode {
      return this.$createElement('span', {
        staticClass: 'v-chip__content'
      }, [
        this.$slots.default,
        this.close && this.genClose()
      ])
    }
  },

  created () {
    if (this.outline) deprecate('outline', 'outlined', this)
  },

  render (h): VNode {
    const data = this.setBackgroundColor(this.color, {
      staticClass: 'v-chip',
      class: this.classes,
      attrs: {
        draggable: this.draggable ? 'true' : undefined,
        tabindex: this.disabled ? -1 : this.$attrs.tabindex
      },
      directives: [
        {
          name: 'show',
          value: this.isActive
        },
        {
          name: 'ripple',
          value: this.ripple != null ? this.ripple : !this.disabled
        }
      ],
      on: this.$listeners
    })

    const color = this.textColor || (this.hasOutline && this.color)

    return h(
      'span',
      this.setTextColor(color, data),
      [this.genContent()]
    )
  }
})
