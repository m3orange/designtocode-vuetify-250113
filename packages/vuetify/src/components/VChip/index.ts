// Styles
import './VChip.sass'

import VChip from './VChip'

export { VChip }
export default VChip
