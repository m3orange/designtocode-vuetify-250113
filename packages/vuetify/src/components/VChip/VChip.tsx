// Styles
import './VChip.sass'

// Components
import { VAvatar } from '@/components/VAvatar'
import { VIcon } from '@/components/VIcon'
import { VChipGroupSymbol } from '@/components/VChipGroup/VChipGroup'

// Composables
import { genOverlays, makeVariantProps, useVariant } from '@/composables/variant'
import { makeBorderProps, useBorder } from '@/composables/border'
import { makeDensityProps, useDensity } from '@/composables/density'
import { makeElevationProps, useElevation } from '@/composables/elevation'
import { makeGroupItemProps, useGroupItem } from '@/composables/group'
import { makeRoundedProps, useRounded } from '@/composables/rounded'
import { makeRouterProps, useLink } from '@/composables/router'
import { makeSizeProps, useSize } from '@/composables/size'
import { makeTagProps } from '@/composables/tag'
import { makeThemeProps, provideTheme } from '@/composables/theme'
import { useProxiedModel } from '@/composables/proxiedModel'

// Directives
import { Ripple } from '@/directives/ripple'

// Utilities
import { defineComponent } from '@/util'

export const VChip = defineComponent({
  name: 'VChip',

  directives: { Ripple },

  props: {
    // TODO: Implement
    // active: Boolean,
    // activeColor: String,
    activeClass: String,
    appendAvatar: String,
    appendIcon: String,
    closable: Boolean,
    closeIcon: {
      type: String,
      default: '$delete',
    },
    closeLabel: {
      type: String,
      default: '$vuetify.close',
    },
    draggable: Boolean,
    filter: Boolean,
    filterIcon: {
      type: String,
      default: '$complete',
    },
    label: Boolean,
    link: Boolean,
    pill: Boolean,
    prependAvatar: String,
    prependIcon: String,
    ripple: {
      type: Boolean,
      default: true,
    },
    text: String,
    modelValue: {
      type: Boolean,
      default: true,
    },

    ...makeBorderProps(),
    ...makeDensityProps(),
    ...makeElevationProps(),
    ...makeGroupItemProps(),
    ...makeRoundedProps(),
    ...makeRouterProps(),
    ...makeSizeProps(),
    ...makeTagProps({ tag: 'span' }),
    ...makeThemeProps(),
    ...makeVariantProps({ variant: 'contained-text' } as const),
  },

  emits: {
    'click:close': (e: Event) => true,
    'update:active': (value: Boolean) => true,
    'update:modelValue': (value: Boolean) => true,
  },

  setup (props, { attrs, emit, slots }) {
    const isActive = useProxiedModel(props, 'modelValue')

    const { themeClasses } = provideTheme(props)
    const { borderClasses } = useBorder(props)
    const { colorClasses, colorStyles, variantClasses } = useVariant(props)
    const { elevationClasses } = useElevation(props)
    const group = useGroupItem(props, VChipGroupSymbol, false)
    const { roundedClasses } = useRounded(props)
    const { sizeClasses } = useSize(props)
    const { densityClasses } = useDensity(props)
    const link = useLink(props, attrs)

    function onCloseClick (e: Event) {
      isActive.value = false

      emit('click:close', e)
    }

    return () => {
      const Tag = (link.isLink.value) ? 'a' : props.tag
      const hasAppend = !!(slots.append || props.appendIcon || props.appendAvatar)
      const hasClose = !!(slots.close || props.closable)
      const hasPrepend = !!(slots.prepend || props.prependIcon || props.prependAvatar)
      const isClickable = !props.disabled && (!!group || link.isClickable.value || props.link)
      const onClickFunc = props.link ? props.link : group?.toggle

      return isActive.value && (
        <Tag
          class={[
            'v-chip',
            {
              'v-chip--disabled': props.disabled,
              'v-chip--label': props.label,
              'v-chip--link': isClickable,
              'v-chip--pill': props.pill,
            },
            themeClasses.value,
            borderClasses.value,
            !group || group.isSelected.value ? colorClasses.value : undefined,
            densityClasses.value,
            elevationClasses.value,
            roundedClasses.value,
            sizeClasses.value,
            variantClasses.value,
            group?.selectedClass.value,
          ]}
          style={[
            !group || group.isSelected.value ? colorStyles.value : undefined,
          ]}
          disabled={ props.disabled || undefined }
          draggable={ props.draggable }
          href={ link.href.value }
          v-ripple={ [isClickable && props.ripple, null] }
          onClick={ isClickable && onClickFunc }
        >
          { genOverlays(isClickable, 'v-chip') }

          { hasPrepend && (
            <div class="v-chip__prepend">
              { slots.prepend
                ? slots.prepend()
                : (
                  <VAvatar
                    icon={ props.prependIcon }
                    image={ props.prependAvatar }
                    size={ props.size }
                  />
                )
              }
            </div>
          ) }

          { slots.default?.({
            isSelected: group?.isSelected.value,
            selectedClass: group?.selectedClass.value,
            select: group?.select,
            toggle: group?.toggle,
            value: group?.value.value,
            disabled: props.disabled,
          }) ?? props.text }

          { hasAppend && (
            <div class="v-chip__append">
              { slots.append
                ? slots.append()
                : (
                  <VAvatar
                    icon={ props.appendIcon }
                    image={ props.appendAvatar }
                    size={ props.size }
                  />
                )
              }
            </div>
          ) }

          { hasClose && (
            <div
              class="v-chip__close"
              onClick={ onCloseClick }
            >
              { slots.close
                ? slots.close({ props: { onClick: onCloseClick } })
                : (
                  <VIcon
                    icon={ props.closeIcon }
                    size="x-small"
                  />
                )
              }
            </div>
          ) }
        </Tag>
      )
    }
  },
})

export type VChip = InstanceType<typeof VChip>
