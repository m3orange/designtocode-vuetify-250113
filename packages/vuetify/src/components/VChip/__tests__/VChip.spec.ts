// Libraries
import Vue from 'vue'

// Plugins
import Router from 'vue-router'

// Components
import VChip from '../VChip'

// Utilities
import {
  createLocalVue,
  mount,
  Wrapper
} from '@vue/test-utils'

describe('VBtn.ts', () => {
  let mountFunction: (options?: object) => Wrapper<Vue>
  let router: Router
  let localVue: typeof Vue

  beforeEach(() => {
    router = new Router()
    localVue = createLocalVue()
    localVue.use(Router)

    mountFunction = (options = {}) => {
      return mount(VChip, {
        localVue,
        router,
        ...options
      })
    }
  })

  it('should have a v-chip class', () => {
    const wrapper = mountFunction()

    expect(wrapper.classes()).toContain('v-chip')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be removable', () => {
    const wrapper = mountFunction({
      propsData: { close: true }
    })

    const close = wrapper.find('.v-chip__close')

    const input = jest.fn(value => wrapper.setProps({ value }))
    wrapper.vm.$on('input', input)

    expect(wrapper.html()).toMatchSnapshot()

    close.trigger('click')
    expect(input).toBeCalledWith(false)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a colored chip', () => {
    const wrapper = mountFunction({
      propsData: {
        color: 'blue',
        textColor: 'green'
      }
    })

    expect(wrapper.element.classList).toContain('blue')
    expect(wrapper.element.classList).toContain('green--text')
  })

  it('should render a clickable chip', () => {
    const wrapper = mountFunction({
      on: {
        click: () => {}
      }
    })

    expect(wrapper.element.classList).toContain('clickable')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a disabled chip', () => {
    const wrapper = mountFunction({
      propsData: {
        disabled: true
      }
    })

    expect(wrapper.element.classList).toContain('v-chip--disabled')

    wrapper.setProps({
      close: true
    })
    expect(wrapper.findAll('.v-chip__close')).toHaveLength(1)
  })

  it('should render a colored outline chip', () => {
    const wrapper = mountFunction({
      propsData: {
        outline: true,
        color: 'blue'
      }
    })

    expect(wrapper.element.classList).toContain('blue')
    expect(wrapper.element.classList).toContain('blue--text')
  })

  it('should render a colored outline chip with text color', () => {
    const wrapper = mountFunction({
      propsData: {
        outline: true,
        color: 'blue',
        textColor: 'green'
      }
    })

    expect(wrapper.element.classList).toContain('blue')
    expect(wrapper.element.classList).toContain('green--text')
  })
})
