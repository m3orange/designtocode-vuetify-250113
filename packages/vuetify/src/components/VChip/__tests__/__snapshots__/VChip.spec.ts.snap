// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VBtn.ts should be removable 1`] = `

<span class="v-chip v-chip--removable theme--light v-size--default">
  <span class="v-chip__content">
    <i aria-hidden="true"
       class="v-icon v-chip__close v-icon--link v-icon--right material-icons theme--light"
    >
      $vuetify.icons.delete
    </i>
  </span>
</span>

`;

exports[`VBtn.ts should be removable 2`] = `

<span class="v-chip v-chip--removable theme--light v-size--default"
      style="display: none;"
>
  <span class="v-chip__content">
    <i aria-hidden="true"
       class="v-icon v-chip__close v-icon--link v-icon--right material-icons theme--light"
    >
      $vuetify.icons.delete
    </i>
  </span>
</span>

`;

exports[`VBtn.ts should have a v-chip class 1`] = `

<span class="v-chip theme--light v-size--default">
  <span class="v-chip__content">
  </span>
</span>

`;
